### PR TITLE
[UI] Update styles & design across the external/internal app screens & components

### PR DESCRIPTION
**Pull request by iwasrobbed**
cc @tonyxiao for casual review (non-blocking)

### CODE REVIEW

File: apps/web/components/common-components.tsx [hunk 1]
```diff
@@ -6,28 +6,29 @@ import type {LayoutProps} from './Layout'
 import {Layout} from './Layout'
 
 export function PageContainer({
-  authenticated = true,
+  requiresAuthentication = true,
```
Comment (iwasrobbed) [inline apps/web/components/common-components.tsx:9]: Note: this was renamed to be explicit about what it does

File: apps/web/contexts/common-contexts.ts [hunk 2]
```diff
@@ -4,6 +4,11 @@ import {createClient} from '@supabase/supabase-js'
 export const supabase = createClient(
   'https://hhnxsazpojeczkeeifli.supabase.co',
   'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhobnhzYXpwb2plY3prZWVpZmxpIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NjAyNjgwOTIsImV4cCI6MTk3NTg0NDA5Mn0.ZDmf1sjsr-UxW2bPgdj3uaqJNUSqkZh8vCB1phn3qqs',
+  {
+    auth: {
+      autoRefreshToken: true,
```
Comment (iwasrobbed) [inline apps/web/contexts/common-contexts.ts:9]: Note: we should default into auto refreshing these (and still need to sanity check people get logged out when it finally expires)

File: apps/web/pages/data.tsx [hunk 3]
```diff
@@ -95,7 +30,7 @@ export default function DataExplorerScreen() {
   // @ts-expect-error
   const apiKey = userInfoRes.data?.apiKey
   // @ts-expect-error
-  const databaseUrl = userInfoRes.data?.databaseUrl
+  const databaseUrl: string = userInfoRes.data?.databaseUrl ?? ''
   // @ts-expect-error
```
Comment (iwasrobbed) [inline apps/web/pages/data.tsx:34]: Lots of VS Code warnings in this class we should try and clean up

File: apps/web/pages/data.tsx [hunk 4]
```diff
@@ -148,24 +77,16 @@ export default function DataExplorerScreen() {
               }}>
               Execute SQL
             </button>
-            <button
-              className="btn"
-              onClick={async () => {
-                // @ts-expect-error
-                const res = await trpcClient.mutation('createDbUser', {})
```
Comment (iwasrobbed) [inline apps/web/pages/data.tsx:155]: @tonyxiao Why was this needed (`create db user`)? Can it be put into a debug menu?

File: apps/web/pages/pipelines.tsx [hunk 5]
```diff
@@ -57,114 +39,164 @@ export default function ConnectionsScreen() {
...
+        </Link>
+      </p>
+    </div>
+  )
+}
+
+// Ledgers UI code (not in MVP for now)
+// TODO: Move this into settings screen (rarely used)
+
+// import {
```
Comment (iwasrobbed) [inline apps/web/pages/pipelines.tsx:151]: Note: ledger UI code is commented out for now; this should get moved into a debug menu somewhere until we're ready to enable it for users

File: apps/web/screens/components/ConnectionCard.tsx [hunk 6]
```diff
@@ -114,12 +110,12 @@ export function ConnectionCard({connection: conn}: ConnectionCardProps) {
                       console.error('syncConnection error', err)
                     })
                 }>
-                <Play />
                 Full sync
               </DropdownMenuItem>
 
+              {/* TODO: Add a confirmation dialog */}
```
Comment (iwasrobbed) [inline apps/web/screens/components/ConnectionCard.tsx:116]: Note: we need a confirmation dialog for deleting a connection

File: apps/web/contexts/common-contexts.ts (hunk 2 above)
Comment (tonyxiao) [reply apps/web/contexts/common-contexts.ts:9]: surprised it's not the supabase default already.

File: apps/web/pages/data.tsx (hunk 4 above)
Comment (tonyxiao) [reply apps/web/pages/data.tsx:155]: yes, it's debug only. I need to implement the logic of creating DB user on demand (e.g. when you access the database url) and also probably via postgres trigger (on auth.users creation)

File: apps/web/screens/components/ConnectionCard.tsx (hunk 6 above)
Comment (tonyxiao) [reply apps/web/screens/components/ConnectionCard.tsx:116]: Yes, and probably language to help you understand what actually happens when you do.

Comment (tonyxiao) [reply apps/web/screens/components/ConnectionCard.tsx:116]: Do we have a task for it?

File: apps/web/components/EnhancedActiveLink.tsx [hunk 7]
```diff
@@ -15,7 +15,7 @@ export const EnhancedActiveLink = React.forwardRef(function ActiveLink(
 ) {
   const router = useRouter()
   const active = router.asPath.startsWith(props.as ?? props.href)
-  const exactActive = (props.as ?? props.href) === router.asPath
```
Comment (tonyxiao) [inline apps/web/components/EnhancedActiveLink.tsx:18]: what was this change about?

File: apps/web/components/Layout.tsx [hunk 8]
```diff
@@ -40,112 +44,115 @@ export function Layout({
...
-                        ? 'btn-primary link-exact-active:btn-disabled'
-                        : 'btn-ghost link-exact-active:text-black',
-                      !l.fixed && 'hidden md:inline-flex',
-                    )}>
-                    {l.label}
-                  </EnhancedActiveLink>
-                ))}
+    <>
+      <Head>
+        <title>Venice - {title}</title>
```
Comment (tonyxiao) [inline apps/web/components/Layout.tsx:50]: This would produce `Venice - Venice - financial data , fast`

File: apps/web/components/Layout.tsx [hunk 9]
```diff
@@ -40,112 +44,115 @@ export function Layout({
...
+      </Head>
+      <div className="relative mx-auto flex h-screen max-w-screen-xl flex-col overflow-y-hidden">
+        <header className="shrink-0 overflow-x-hidden">
+          {/* TODO: Add global control for envName as well as currentUserId in developer mode */}
```
Comment (tonyxiao) [inline apps/web/components/Layout.tsx:54]: yes, it would be nice to wireframe how you envision the developer console to look like.

File: apps/web/contexts/common-contexts.ts [hunk 10]
```diff
@@ -4,6 +4,11 @@ import {createClient} from '@supabase/supabase-js'
 export const supabase = createClient(
   'https://hhnxsazpojeczkeeifli.supabase.co',
```
Comment (tonyxiao) [inline apps/web/contexts/common-contexts.ts:5]: Thought you added an env var for supabase url?

File: apps/web/global.css [hunk 11]
```diff
@@ -2,28 +2,6 @@
 @import 'tailwindcss/components';
 @import 'tailwindcss/utilities';
 
-@layer base {
```
Comment (tonyxiao) [inline apps/web/global.css:5]: what's our new font choice?

File: apps/web/pages/auth.tsx [hunk 12]
```diff
@@ -1,18 +1,46 @@
...
+        </div>
       </Container>
     </PageContainer>
   )
 }
+
+const veniceThemeTweaks = {
+  default: {
+    colors: {
+      brand: '#12B886',
```
Comment (tonyxiao) [inline apps/web/pages/auth.tsx:39]: should probably import from the theme file?

File: apps/web/styles/themes.js [hunk 13]
```diff
@@ -0,0 +1,101 @@
+/* Venice's Core Theme */
```
Comment (tonyxiao) [inline apps/web/styles/themes.js:1]: why is this file `.js` when all others are `.ts`?

File: apps/web/screens/components/ConnectionCard.tsx (hunk 6 above)
Comment (iwasrobbed) [reply apps/web/screens/components/ConnectionCard.tsx:116]: Yes, I'll take this one

File: apps/web/components/EnhancedActiveLink.tsx (hunk 7 above)
Comment (iwasrobbed) [reply apps/web/components/EnhancedActiveLink.tsx:18]: `router.asPath` includes query params so it wasn't matching the link and highlighting the nav links like intended. Using `.pathname` strips the query params

File: apps/web/components/Layout.tsx (hunk 8 above)
Comment (iwasrobbed) [reply apps/web/components/Layout.tsx:50]: True, will fix

File: apps/web/components/Layout.tsx (hunk 9 above)
Comment (iwasrobbed) [reply apps/web/components/Layout.tsx:54]: Let's discuss the dev use cases soon

File: apps/web/contexts/common-contexts.ts (hunk 10 above)
Comment (iwasrobbed) [reply apps/web/contexts/common-contexts.ts:5]: In another branch, but it was broken on every env except prod so I didn't roll it in for now. Feel free to tweak or I will soon

File: apps/web/global.css (hunk 11 above)
Comment (iwasrobbed) [reply apps/web/global.css:5]: Matches the venice.is site:
- Montserrat for primary
- some monospace (e.g. Source Code Pro is fine) for code or technical areas

File: apps/web/styles/themes.js (hunk 13 above)
Comment (iwasrobbed) [reply apps/web/styles/themes.js:1]: Compiler seems to barf when importing it into tailwind's config file as ts while also importing into tsx files as a module. You'll have to show me
